cpp/src/arrow/compute/kernels/hash_aggregate.cc

```diff
@@ -1786,14 +1830,13 @@ struct GroupedMinMaxImpl : public GroupedAggregator {
   }
 
   Status Consume(const ExecBatch& batch) override {
-    auto raw_mins = reinterpret_cast<CType*>(mins_.mutable_data());
-    auto raw_maxes = reinterpret_cast<CType*>(maxes_.mutable_data());
+    auto raw_mins = reinterpret_cast<ArrType*>(mins_.mutable_data());
+    auto raw_maxes = reinterpret_cast<ArrType*>(maxes_.mutable_data());
```

Comment:
Since `mins_` and `maxes_` are TypedBufferBuilders, is the `reinterpret_cast` at all required?